status-im/status-desktop
# Refactor/16286 refactor profile context menu

closes https://github.com/status-im/status-desktop/issues/16286

### What does the PR do

This refactor ProfileContextMenu to make it a functional component by:
- refactored out direct calls to backend, and passing backend data structures and moved this logic to the callers, also refactored common calls between the callers
- common types of context menus have been extracted to their sub components which removes a lot of logic too and makes the behaviour very clear
- user verification workflow (which was already disabled) has been removed

### Affected areas

Any where in which one can right click or click on `...` menu to see the profile context menu, to my knoweldge this includes:
- left click or right click username on a message in all chats (1 on 1, group chats and communities)
- right click username on a member list of a group chat
- right click username on a member list of a community channel
- clicking on the `...` menu in contacts list in settings

### Impact on end user

The before/after behaviour should be the same for the end user

### How to test

- test affected areas and that 1) the profile context menu opens correctly with the right menu items visible) and 2) menu items works as expected (e.g clicking "add nickname" should open the add nickname modal)

note: The goal here is to refactor under the hood while keeping the same behaviour as before. For example, if a profile is in a blocked state, the trustworthy/untrustworthy mark doesn't show because it didn't show before either, if that behaviour should change is beyond the scope of this PR.

### Risk 

Needs further testing before merging:
- [ ] contacts with ENS, I couldn't verify if this works as expected because I keep seeing the ring (meaning the ENS verification didn't work) it's not necessary a issue with this PR but needs to be checked
- [ ] any other areas that might have been forgotten, I haven't tested in a contact list in a community admin settings
